enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,362 @@
+alternative will be replacing `nc` w/ a simple binary named `konnectivity-nc`
+that supports both mTLS https or grpc proxy as an extension.
+
+An very simple revision of `konnectiviy-nc` will be:
+
+[https://gist.github.com/yue9944882/3266bb3a73fde5538e861ae174f6f177](https://gist.github.com/yue9944882/3266bb3a73fde5538e861ae174f6f177)
+
+And the ssh command will be:
+
+```shell
+ssh user@host \
+-o "ProxyCommand=konnectivity-nc \
+--proxy-ca-cert <filepath to ca cert> \
+--proxy-cert <filepath to x509 cert> \
+--proxy-key <filepath to private key> \
+%h %p"
+```
+
+#### Framework Integration
+
+Based on the extensibility provided by the addon-framework, the new proxy addon
+will be implementing the required interfaces in the following behavior:
+
+
+```go
+var _ AgentAddon = &ApiserverNetworkProxyAddon{}
+
+func (a *ApiserverNetworkProxyAddon) Manifests(cluster *clusterv1.ManagedCluster, config runtime.Object) ([]runtime.Object, error) {
+    // 1. a service-account with fundamental spoke cluster permissions.
+    // 2. a deployment installing the proxy-agents
+}
+
+func (a *ApiserverNetworkProxyAddon) GetAgentAddonOptions() AgentAddonOptions {
+	// using the `ProxyConfiguration` above as options
+}
+
+var _ AgentAddonWithRegistration = &ApiserverNetworkProxyAddon{}
+
+func (a *ApiserverNetworkProxyAddon) HubRBAC(cluster *clusterv1.ManagedCluster, group string) (*rbacv1.Role, *rbacv1.RoleBinding) {
+	// none
+}
+
+func (a *ApiserverNetworkProxyAddon) BootstrapKubeConfig(cluster *clusterv1.ManagedCluster) ([]byte, error) {
+	// a bootstrap kubeconfig for raising initial CSR requests
+}
+
+
+func (a *ApiserverNetworkProxyAddon) RegistrationConfig(cluster *clusterv1.ManagedCluster) []RegistrationConfig {
+	// groups:  ["open-cluster-management:proxy-agents"]
+}
+
+func (a *ApiserverNetworkProxyAddon) CSRApproveCheck(cluster *clusterv1.ManagedCluster, csr *certificatesv1.CertificateSigningRequest) bool {
+	// matching all those CSR with the expected bootstrap kubeconfig
+}
+```
+
+## Future Work
+
+#### An Optional Aggregated-Apiserver Gateway
+
+In order to simplify the direct multi-cluster api calls further, we plan to
+make the addon controller to opt-in to install an aggregated-apiserver which
+provides a `/proxy` subresource that allows users to access the managedclusters
+via plain HTTPS calls w/o proxy. An example will be:
+
+```bash
+# probing managed cluster's /healthz
+kubectl get --raw="/apis/<the addon api-group>/<the addon api-version>/<addon-resource-name>/<managed cluster name>/proxy/healthz"
+# listing managed cluster's namespaces
+kubectl get --raw="/apis/<the addon api-group>/<the addon api-version>/<addon-resource-name>/<managed cluster name>/proxy/api/v1/namespaces"
+```
+
+And the official golang and java SDKs are supposed to talk to the managed cluster
+by simply adding an api prefix to the requests.
+
+#### Proxy Server Authorization 
+
+Even if the proxy-servers are mTLS-secured, all the client talking to the servers
```

Comment:
I think one option is we can transparently pass the credential of the request to the spoke, if all spoke are at the same id provider.

Reply:
@qiujian16 thats exactly now `agentAuthentication` works we talked above. i will eliminate this section.